# Mid-term-project | ML-ZOOM-CAMP | predict-term deposit

List of contents:

1. Problem description
2. Getting the dataset
3. Reading the dataset with pandas

## 1. Problem Description 

## 2. Getting the dataset

- link to dataset: [https://www.kaggle.com/datasets/aslanahmedov/predict-term-deposit](https://www.kaggle.com/datasets/aslanahmedov/predict-term-deposit)

or,

[https://raw.githubusercontent.com/bhasarma/mlcoursezoom-camp/main/WK08-09-midterm-project/predict-term-deposit-data.csv](https://raw.githubusercontent.com/bhasarma/mlcoursezoom-camp/main/WK08-09-midterm-project/predict-term-deposit-data.csv)

You can download the dataset into your local directory with `wget` 

In [1]:
data = 'https://raw.githubusercontent.com/bhasarma/mlcoursezoom-camp/main/WK08-09-midterm-project/predict-term-deposit-data.csv'

In [2]:
#!wget $data  #uncomment it, if you haven't downoaded data already.

In [3]:
ls

notebook.ipynb  predict-term-deposit-data.csv  README.md  report.md


## 3. Reading the dataset with pandas 

In [15]:
%matplotlib inline

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [6]:
df = pd.read_csv('predict-term-deposit-data.csv')

In [10]:
df.head().T

0           1             2            3        4
Id               1001        1002          1003         1004     1005
age             999.0        44.0          33.0         47.0     33.0
job        management  technician  entrepreneur  blue-collar  unknown
marital       married      single       married      married   single
education    tertiary   secondary     secondary      unknown  unknown
default            no          no            no           no       no
balance        2143.0        29.0           2.0       1506.0      1.0
housing           yes         yes           yes          yes       no
loan               no          no           yes           no       no
contact       unknown     unknown       unknown      unknown  unknown
day                 5           5             5            5        5
month             may         may           may          may      may
duration          261         151            76           92      198
campaign            1           1             1            1        1
pdays              -1          -1            -1           -1       -1
previous            0           0             0            0        0
poutcome      unknown     unknown       unknown      unknown  unknown
y                  no          no            no           no       no

In [11]:
df.shape

(45211, 18)

In [14]:
df.columns

Index(['Id', 'age', 'job', 'marital', 'education', 'default', 'balance',
       'housing', 'loan', 'contact', 'day', 'month', 'duration', 'campaign',
       'pdays', 'previous', 'poutcome', 'y'],
      dtype='object')

In [16]:
df.dtypes

Id             int64
age          float64
job           object
marital       object
education     object
default       object
balance      float64
housing       object
loan          object
contact       object
day            int64
month         object
duration       int64
campaign       int64
pdays          int64
previous       int64
poutcome      object
y             object
dtype: object

In [18]:
df.describe()

Id           age        balance           day      duration  \
count  45211.000000  45202.000000   45208.000000  45211.000000  45211.000000   
mean   23606.000000     40.954714    1362.346620     15.806419    258.163080   
std    13051.435847     11.539144    3044.852387      8.322476    257.527812   
min     1001.000000     -1.000000   -8019.000000      1.000000      0.000000   
25%    12303.500000     33.000000      72.000000      8.000000    103.000000   
50%    23606.000000     39.000000     448.000000     16.000000    180.000000   
75%    34908.500000     48.000000    1428.000000     21.000000    319.000000   
max    46211.000000    999.000000  102127.000000     31.000000   4918.000000   

           campaign         pdays      previous  
count  45211.000000  45211.000000  45211.000000  
mean       2.763841     40.197828      0.580323  
std        3.098021    100.128746      2.303441  
min        1.000000     -1.000000      0.000000  
25%        1.000000     -1.000000      0.000000  
50%        2.000000     -1.000000      0.000000  
75%        3.000000     -1.000000      0.000000  
max       63.000000    871.000000    275.000000

In [23]:
df.isnull().sum()

Id           0
age          9
job          0
marital      0
education    0
default      0
balance      3
housing      0
loan         0
contact      0
day          0
month        0
duration     0
campaign     0
pdays        0
previous     0
poutcome     0
y            0
dtype: int64

In [24]:
df.nunique()

Id           45211
age             79
job             12
marital          3
education        4
default          2
balance       7168
housing          2
loan             2
contact          3
day             31
month           12
duration      1573
campaign        48
pdays          559
previous        41
poutcome         4
y                2
dtype: int64

In [25]:
for col in df.columns:
    print(col)
    print(df[col].unique()[:5])
    print(df[col].nunique())
    print()

Id
[1001 1002 1003 1004 1005]
45211

age
[999.  44.  33.  47.  35.]
79

job
['management' 'technician' 'entrepreneur' 'blue-collar' 'unknown']
12

marital
['married' 'single' 'divorced']
3

education
['tertiary' 'secondary' 'unknown' 'primary']
4

default
['no' 'yes']
2

balance
[2.143e+03 2.900e+01 2.000e+00 1.506e+03 1.000e+00]
7168

housing
['yes' 'no']
2

loan
['no' 'yes']
2

contact
['unknown' 'cellular' 'telephone']
3

day
[5 6 7 8 9]
31

month
['may' 'jun' 'jul' 'aug' 'oct']
12

duration
[261 151  76  92 198]
1573

campaign
[1 2 3 5 4]
48

pdays
[ -1 151 166  91  86]
559

previous
[0 3 1 4 2]
41

poutcome
['unknown' 'failure' 'other' 'success']
4

y
['no' 'yes']
2



A summary about the data:

1. We have 18 features (or, variables or, columns) and 45211 columns. 